crates/core/src/factories/core_fetch.rs

```diff
@@ -335,6 +391,54 @@ impl CoreFetch {
         }
     }
 
+    async fn spawn_response_task(
+        response_rx: Arc<tokio::sync::Mutex<Receiver<FetchResponse>>>,
+        peer_store: peer_store::DynPeerStore,
+        op_store: op_store::DynOpStore,
+        transport: DynTransport,
+        space_id: SpaceId,
+    ) {
+        while let Some((op_ids, agent_id)) =
+            response_rx.lock().await.recv().await
+        {
+            tracing::debug!(?op_ids, ?agent_id, "incoming request");
+            let peer = match CoreFetch::get_peer_url_from_store(
+                &agent_id,
+                peer_store.clone(),
+            )
+            .await
+            {
+                None => continue,
```

Comment:
A comment here might be nice, I think it's pretty clear but something like the one below for the `is_empty` check would be good